src/internal/mod.rs

```diff
@@ -477,6 +486,51 @@ impl<'de> Deserialize<'de> for PrivateKey {
     }
 }
 
+impl PrivateKey {
+    /// Augment this private key with another, producing a new PrivateKey
+    fn augment(&self, augmenting_key: &AugmentationFactor) -> Result<PrivateKey, IronOxideErr> {
+        use recrypt::Revealed;
+        let zero: RecryptPrivateKey = RecryptPrivateKey::new([0u8; 32]);
+        if Revealed((augmenting_key.clone().0).0) == Revealed(zero.clone()) {
+            Err(IronOxideErr::UserPrivateKeyRotationError(
+                "Augmenting key cannot be zero".into(),
+            ))
+        } else {
+            // this subtraction needs to be the additive inverse of what the service is doing
```

Comment:
I thought I had mentioned this, but maybe that was somewhere else. Instead of doing the subtraction then comparing the result to zero, you could compare the augmenting key with self.0. If they are equal, it will end up failing with a zero value.

Reply:
This is somewhat worse now as it it will require another clone of the `self` value, but it allows eliminating a level of nesting which is nice. The performance impact of the clone will go away when we fix Recrypt's `Revealed` type.

Reply:
I think it's good to eliminate the subtraction if it is easy to do.

Reply:
Maybe I'm still not understanding what you are proposing. The subtraction isn't eliminated it's simply delayed. If the augmenting value were the same as self, it would be eliminated, but that is expected to be incredibly rare.

See https://github.com/IronCoreLabs/ironoxide/pull/59/files#diff-b17bedabf75c7285b05afbf395928056R491 for my implementation of delaying the subtraction.

Reply:
You're right. Working to eliminate something that will probably never happen was not worth thinking about.